packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/impactanalysis/InvokeImpactAnalysisToolEventHandler.java

```diff
@@ -0,0 +1,141 @@
+        this.representationSearchService = Objects.requireNonNull(representationSearchService);
+        this.toolExecutors = Objects.requireNonNull(toolExecutors);
+
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+    }
+
+    @Override
+    public boolean canHandle(IEditingContext editingContext, IInput input) {
+        return input instanceof InvokeImpactAnalysisToolInput;
+    }
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IInput input) {
+        this.counter.increment();
+
+        String message = this.messageService.invalidInput(input.getClass().getSimpleName(), InvokeImpactAnalysisToolInput.class.getSimpleName());
+        IPayload payload = new ErrorPayload(input.id(), message);
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, editingContext.getId(), input);
+
+        if (input instanceof InvokeImpactAnalysisToolInput invokeImpactAnalysisToolInput && editingContext instanceof EditingContext siriusEditingContext) {
+            ResourceSet resourceSet = siriusEditingContext.getDomain().getResourceSet();
+            var snapshotBeforeToolExecution = resourceSet.getResources().stream().map(resource -> this.resourceToDocumentService.toDocument(resource, false))
+                    .filter(Optional::isPresent)
+                    .map(Optional::get).toList();
+
+            ChangeRecorder changeRecorder = siriusEditingContext.getChangeRecorder();
+
+            var representationOptional = this.representationSearchService.findById(editingContext, invokeImpactAnalysisToolInput.representationId());
+            if (representationOptional.isPresent()) {
+                changeRecorder.beginRecording(siriusEditingContext.getDomain().getResourceSet().getResources());
+
+                IStatus toolExecutionResult = this.toolExecutors.stream().filter(executor -> executor.canExecute(editingContext, representationOptional.get())).findFirst()
+                        .map(executor -> executor.execute(editingContext, representationOptional.get(), invokeImpactAnalysisToolInput.toolId(), invokeImpactAnalysisToolInput.targetObjectId(),
+                                invokeImpactAnalysisToolInput.variables()))
+                        .orElse(new Failure(""));
+
+                var diff = siriusEditingContext.getChangeRecorder().summarize();
+                siriusEditingContext.getChangeRecorder().endRecording();
+                resourceSet.getResources().clear();
+                for (var documentSnapshot : snapshotBeforeToolExecution) {
+                    this.resourceLoader.toResource(resourceSet, documentSnapshot.document().getId().toString(), documentSnapshot.document().getName(), documentSnapshot.document().getContent(), false);
+                }
```

Comment:
Loading an `IEditingContext` involves more than that, it will not produce the same state as the `IEditingContextSearchService` would.